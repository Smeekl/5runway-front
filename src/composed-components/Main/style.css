.main {
  padding: 168px 165px 0 165px;
  position: relative;
}

.find-aircraft {
  min-height: 540px;
}

.find-aircraft__title {
  font-weight: 900;
  font-size: 58px;
  color: var(--color-primary);
  margin: 0 0 2px 0;
}

.find-aircraft__subtitle {
  font-size: 18px;
  font-weight: 300;
  width: 255px;
}

.find-aircraft__search {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  position: relative;
  width: 470px;
}

.find-aircraft__field {
  width: 349px;
  height: 40px;
  border: 1px solid var(--color-secondary-mad-blue);
  border-radius: 15px;
}

.find-aircraft__btn {
  width: 168px;
  height: 41px;
  background: var(--color-primary-blue);
  border-radius: 15px;
  position: absolute;
  top: 0;
  right: 0;
  transition: 0.3s all;
  z-index: 2;
}

.find-aircraft__btn > span {
  font-size: 16px;
  color: var(--color-secondary-white);
}

.find-aircraft__btn:hover,
.find-aircraft__btn:focus {
  background: var(--color-primary);
  border: 0;
}

.find-aircraft__background {
  z-index: -1;
  background: url(../../assets/mainPage/helicopterBackground.svg) no-repeat;
  background-position: -10px -85px;
  position: absolute;
  top: 35px;
  right: 0;
  min-height: 800px;
}

.main__categories {
  min-height: 906px;
}

.main__top-products {
  min-height: 1337px;
}

.main__articles {
  min-height: 793px;
}
