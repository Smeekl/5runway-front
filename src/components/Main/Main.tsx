import './style.css';

const Main = () => (
  <div>
    <h1>Main</h1>
  </div>
);

export default Main;
