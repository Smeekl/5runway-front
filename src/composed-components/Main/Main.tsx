import helicopter from '@mainPageIcons/helicopter.png';
import Button from '@components/Button/Button';
import SearchInput from '@components/SearchInput/SearchInput';
import './style.css';

const Main = () => (
  <main className="main">
    <div className="main__find-aircraft find-aircraft">
      <h1 className="find-aircraft__title">Find your AirCraft</h1>
      <p className="find-aircraft__subtitle">
        A huge selection of aviation equipment
      </p>
      <div className="find-aircraft__search">
        <SearchInput startIcon={true} />
        <Button className="find-aircraft__btn">Search</Button>
      </div>
      <div className="find-aircraft__background">
        <img className="find-aircraft__image" src={helicopter} />
      </div>
    </div>
    <div className="main__categories">
      <h2>Categories</h2>
    </div>
    <div className="main__top-products">
      <h2>Top products</h2>
    </div>
    <div className="main__articles">
      <h2>Articles</h2>
    </div>
  </main>
);

export default Main;
