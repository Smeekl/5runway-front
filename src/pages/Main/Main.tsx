import Main from '@composed-components/Main/Main';
import Layout from '@layouts/Layout/Layout';

const MainPage = () => (
  <Layout>
    <Main />
  </Layout>
);

export default MainPage;
