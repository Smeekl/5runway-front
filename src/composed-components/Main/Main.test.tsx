import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Main from './Main';

describe('Main UI tests', () => {
  test('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Main />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should be in the document', () => {
    const { container } = render(
      <MemoryRouter>
        <Main />
      </MemoryRouter>
    );
    expect(container).toBeInTheDocument();
  });
});
