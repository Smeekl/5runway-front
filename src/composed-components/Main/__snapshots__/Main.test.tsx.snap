// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main UI tests should render correctly 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="main__find-aircraft find-aircraft"
    >
      <h1
        class="find-aircraft__title"
      >
        Find your AirCraft
      </h1>
      <p
        class="find-aircraft__subtitle"
      >
        A huge selection of aviation equipment
      </p>
      <div
        class="find-aircraft__search"
      >
        <span
          class="ant-input-affix-wrapper search__field"
        >
          <span
            class="ant-input-prefix"
          >
            <reactcomponent
              class="search__icon"
            />
          </span>
          <input
            class="ant-input"
            type="text"
            value=""
          />
        </span>
        <button
          class="ant-btn ant-btn-default find-aircraft__btn"
          type="button"
        >
          <span>
            Search
          </span>
        </button>
      </div>
      <div
        class="find-aircraft__background"
      >
        <img
          class="find-aircraft__image"
          src="test-file-stub"
        />
      </div>
    </div>
    <div
      class="main__categories"
    >
      <h2>
        Categories
      </h2>
    </div>
    <div
      class="main__top-products"
    >
      <h2>
        Top products
      </h2>
    </div>
    <div
      class="main__articles"
    >
      <h2>
        Articles
      </h2>
    </div>
  </main>
</div>
`;
